streaming-binlog/src/main/java/org/openmrs/analytics/FhirConverter.java

```diff
@@ -106,8 +117,16 @@ GeneralConfiguration getEventsToFhirConfig(String fileName) throws IOException {
 		Gson gson = new Gson();
 		Path pathToFile = Paths.get(fileName);
 		try (Reader reader = Files.newBufferedReader(pathToFile.toAbsolutePath(), StandardCharsets.UTF_8)) {
-			GeneralConfiguration generalConfig = gson.fromJson(reader, GeneralConfiguration.class);
-			return generalConfig;
+			GeneralConfiguration generalConfiguration = gson.fromJson(reader, GeneralConfiguration.class);
+			return generalConfiguration;
 		}
 	}
+	
+	private HashMap<String, EventConfiguration> mapOutEventsToFhirConfig(GeneralConfiguration generalConfig) {
```

Comment:
Suggestion: It seems that `GeneralConfiguration` is not really needed anywhere and instead the map version is really what we need. So given that we have full control on config now, I think it is better to create something like an `EventConfigMap` class which maintains a map from table names to `EventConfiguration`. Inside `EventConfigMap` you can have a private inner class `GeneralConfiguration` which is only used for reading the json file (and its content is converted to the map right away). Maybe `EventConfigMap` can also be an inner class of `EventConfigMap`. WDYT?

Reply:
That was actually my initial intention but remembered we have about 15+ debezium parameters that can be modeled in the generalConfiguration.  I.e we have a generalConfig model that houses 

- EventToFhir map
- DebeziumParams
- Other future pipeline params

That we can have one jsonschema with all necessary pipeline properties i.e 
```
//generalConfig components
{
	"debeziumConfig": { 15+ params},
	"eventConfigurations": [],
}
```
WDYT?

Reply:
I see; the general idea of moving those Debezium config params to the same json file sounds good to me but still I think you need a layer of abstraction around `GeneralConfiguration`. I mean take the map that you build around this, I prefer that we provide that API directly in the config class instead of creating the map here.

Another idea, if you prefer to have the same json config exposed as public API (e.g., as in `GeneralConfiguration`) is to create a map in the json file instead of the current array. That way you don't need to create the map here. Either way is fine with me.

Reply:
Nice idea on using a map instead of array, I have implemented this suggestion, and will push shortly!